os/fs/tmpfs/fs_tmpfs.c

```diff
@@ -1720,6 +1722,53 @@ static int tmpfs_fstat(FAR const struct file *filep, FAR struct stat *buf)
 	return OK;
 }
 
+/****************************************************************************
+ * Name: tmpfs_truncate
+ ****************************************************************************/
+
+static int tmpfs_truncate(FAR struct file *filep, off_t length)
+{
+	FAR struct tmpfs_file_s *tfo;
+	size_t oldsize;
+	int ret = OK;
+
+	fvdbg("filep: %p length: %ld\n", filep, (long)length);
+	DEBUGASSERT(filep != NULL && length >= 0);
+
+	/* Recover our private data from the struct file instance */
+	tfo = filep->f_priv;
+
+	/* Get exclusive access to the file */
+	tmpfs_lock_file(tfo);
+
+	/* Get the old size of the file.  Do nothing if the file size is not
+	 * changing.
+	 */
+	oldsize = tfo->tfo_size;
+	if (oldsize != length) {
+		/* The size is changing.. up or down.  Reallocate the file memory. */
+		ret = tmpfs_realloc_file(&tfo, (size_t)length);
+		if (ret < 0) {
+			goto errout_with_lock;
+		}
+
+		filep->f_priv = tfo;
+
+		/* If the size has increased, then we need to zero the newly added
+		 * memory.
+		 */
+		if (length > oldsize) {
+			memset(&tfo->tfo_data[oldsize], 0, length - oldsize);
```

Comment:
'\0' is correct. actually '0' works in TizenRT but \0 is correct value. it depend on compiler.